build.gradle

```
@@ -154,23 +154,72 @@ modrinth {
     String projectVersion = project.version.toString()
     List<String> projectVersionNumber = projectVersion.split("-", 2)
     String projectVersionName = "Release ${projectVersionNumber[0]}"
-    if (projectVersion.contains("beta")) {
-        List<String> projectBeta = projectVersionNumber[1].split("[.]", 2)
+    if (projectVersion.contains("patch")) {
+        List<String> _projectVersionNumber = projectVersion.split("\\+", 2)
+        List<String> projectPatch = _projectVersionNumber[1].split("\\.", 2)
+        projectVersionName = "${_projectVersionNumber[0]} - Patch ${projectPatch[1]}"
+    } else if (projectVersion.contains("rc")) {
+        List<String> projectRC = projectVersionNumber[1].split("\\.", 2)
+        projectVersionName = "${projectVersionNumber[0]} - Release Candidate ${projectRC[1]}"
+        versionType = "beta"
+    } else if (projectVersion.contains("beta")) {
+        List<String> projectBeta = projectVersionNumber[1].split("\\.", 2)
         projectVersionName = "${projectVersionNumber[0]} - Beta ${projectBeta[1]}"
         versionType = "beta"
     } else if (projectVersion.contains("alpha")) {
-        List<String> projectAlpha = projectVersionNumber[1].split("[.]", 2)
+        List<String> projectAlpha = projectVersionNumber[1].split("\\.", 2)
         projectVersionName = "${projectVersionNumber[0]} - Alpha ${projectAlpha[1]}"
         versionType = "alpha"
-    } else if (projectVersion.contains("rc")) {
-        List<String> projectRC = projectVersionNumber[1].split("[.]", 2)
-        projectVersionName = "${projectVersionNumber[0]} - Release Candidate ${projectRC[1]}"
-        versionType = "beta"
     }
 
     token = System.getenv("MODRINTH_TOKEN")
     projectId = "CmHD69Pj"
     versionName = projectVersionName
-    changelog = "The changelog can be seen at https://github.com/twitch4j/twitch4j/releases/tag/v${project.version}"
+    changelog = "### The Twitch4J changelog can be seen at https://github.com/twitch4j/twitch4j/releases/tag/v${project.version}\n\n" + CHANGELOG
+    gameVersions = minecraftVersions
     uploadFile = remapJar
 }
+
+import groovy.json.JsonSlurper
+import groovy.xml.XmlSlurper
+
+import java.nio.charset.StandardCharsets
+
+task updateMinecraftVersions {
+    group = "update"
+    doFirst {
+        def json = new URL("https://meta.fabricmc.net/v2/versions/game").text
+        def versions = new JsonSlurper().parseText(json)
+        def output = new StringBuilder()
+        def i = 0
+        for (def version : versions) {
+            if (version.stable) {
+                if (i > 0) {
+                    output.append("\n")
+                }
+                output.append("${version.version}")
+                i++
+            }
+        }
```

Would this work or would `readLines()` include an empty string at the end?
```suggestion
        for (def version : versions) {
            if (version.stable) {
                output.append("${version.version}")
                output.append("\n")
            }
        }
```

---

I imagine so, this is a code I extracted from Fabric Language Kotlin

---

It does not

---

Change included on [chore: Improve build.gradle](https://github.com/twitch4j/twitch4j-fabric/pull/1/commits/8fe0b0d5f88998af1c9b8f26f2cd1c533c741d67)